.github/CODEOWNERS

```diff
@@ -15,6 +15,8 @@
 /.github/workflows @NixOS/Security @Mic92 @zowoq
 /.github/workflows/check-nix-format.yml @infinisil
 /.github/workflows/nixpkgs-vet.yml @infinisil @philiptaron
+/.github/workflows/codeowners.yml @infinisil
+/.github/OWNERS @infinisil
 /ci @infinisil @philiptaron @NixOS/Security
 
```

Comment:
> This PR still runs the native CODEOWNERS together with the alternative mechanism, so that we can run it for some time to confirm that it works correctly. Once confirmed, we'll be able to easily turn off the native CODEOWNERS and just rely on the alternate mechanism.

I think it may still be worthwhile to keep the native codeowners file around for specific parts of the codebase, that way we can limit who can merge using branch rules on specific parts that could be considered critical.

Of course that kind usage should probably be limited, and wouldn't help with avoiding pings for those parts.

Reply:
Also, for what it's worth, validate codeowners could probably be contorted slightly to support failing on certain files we specify, and then have a CI step that could be required with a GitHub branch rule to block merge, but that would then need to be overridden either by someone with correct permissions or some more complicated mechanism.

Reply:
Yeah there's a bunch more things we could do here. In particular I'm very interested in implementing local declaration of code ownership, e.g. using

```nix
/* Owner: @infinisil */
null
```

This could then be extended to support specifying required approvals or more:

```nix
/*
  Owner: @infinisil
  NeedsToApprove: @cafkafk 
*/
null
```

Or we could even do something like: Request a review by the maintainer, but if they don't respond within a week, request a review by the maintainer of the parent directory, etc.

Of course this all needs to be implemented unless we can find a third-party project that did it already, but the possibilities are almost endless :)

Reply:
```
/*
  Owner: @infinisil
  NeedsToApprove: @cafkafk 
*/
```

To workshop a bit more: why would an NTA not be an owner and vice-versa?

Reply:
I'd say one would implicitly be added to the other. I guess expressing it like
```
/*
  Owners: @infinisil @cafkafk (approval-needed)
*/
```

would make this less confusing